src/Controls/src/BindingSourceGen/PathParser.cs

```diff
@@ -162,7 +171,9 @@ internal PathParser(GeneratorSyntaxContext context, bool enabledNullable)
         }
 
         var name = GetIndexerName(elementAccessSymbol);
-        IPathPart part = new IndexAccess(name, indexValue);
+        var isReferenceType = typeSymbol?.IsReferenceType ?? false;
+        var isNullableValueType = typeSymbol != null ? BindingGenerationUtilities.IsNullableValueType(typeSymbol) : false;
```

Comment:
There's quite a lot of code duplication going on here and it's getting hard to read. We should consider moving this to a separate method at least.